src/main/java/FileEncryptor/FileEncryptor.java

```diff
@@ -29,20 +29,35 @@ public FileEncryptor(EncryptionAlgorithm algorithm, String pathToFile) throws IO
         }
     }
 
-    public void encryptFile() throws IOException {
+    public void encryptFile() throws IOException, NullPointerException {
         String textToEncrypt = fileToEncrypt.readFromFile();
-        Key key = new Key(nameOfAlgorithm);
+        Key key = new Key();
         String encryptedData = algorithm.encryptFile(textToEncrypt, key.getKey());
         encryptedFile.writeToFile(encryptedData);
         keyFile.writeToFile(Integer.toString(key.getKey()));
-        if (nameOfAlgorithm == "DoubleEncryption"){
-        }
     }
 
-    public void decryptFile() throws IOException {
+    public void decryptFile() throws IOException, NullPointerException {
         int key = Integer.parseInt(keyFile.readFromFile().strip());
         String dataToDecrypt = encryptedFile.readFromFile();
         String decryptedData = algorithm.decryptFile(dataToDecrypt, key);
         decryptedFile.writeToFile(decryptedData);
     }
+
+    public String getPathToDecryptedFile(){
+        return this.decryptedFile.getPathToFile();
+    }
+
```

Comment:
why not to use comparable interface and compareTo function in FileEncryptor class ( not inner class)?